src/ripple/app/paths/PathRequest.cpp

```diff
@@ -279,65 +267,87 @@ Json::Value PathRequest::doCreate (
         }
     }
 
-    valid = bValid;
     return status;
 }
 
-int PathRequest::parseJson (Json::Value const& jvParams, bool complete)
+int PathRequest::parseJson (Json::Value const& jvParams)
 {
-    int ret = PFR_PJ_NOCHANGE;
+    if (! jvParams.isMember(jss::source_account))
+    {
+        jvStatus = rpcError(rpcSRC_ACT_MISSING);
+        return PFR_PJ_INVALID;
+    }
 
-    if (jvParams.isMember (jss::source_account))
+    if (! jvParams.isMember(jss::destination_account))
     {
-        raSrcAccount = parseBase58<AccountID>(
-            jvParams[jss::source_account].asString());
-        if (! raSrcAccount)
-        {
-            jvStatus = rpcError (rpcSRC_ACT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError(rpcDST_ACT_MISSING);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    if (! jvParams.isMember(jss::destination_amount))
     {
-        jvStatus = rpcError (rpcSRC_ACT_MISSING);
+        jvStatus = rpcError(rpcDST_AMT_MISSING);
         return PFR_PJ_INVALID;
     }
 
-    if (jvParams.isMember (jss::destination_account))
+    raSrcAccount = parseBase58<AccountID>(
+        jvParams[jss::source_account].asString());
+    if (! raSrcAccount)
     {
-        raDstAccount = parseBase58<AccountID>(
-            jvParams[jss::destination_account].asString());
-        if (! raDstAccount)
-        {
-            jvStatus = rpcError (rpcDST_ACT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError (rpcSRC_ACT_MALFORMED);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    raDstAccount = parseBase58<AccountID>(
+        jvParams[jss::destination_account].asString());
+    if (! raDstAccount)
     {
-        jvStatus = rpcError (rpcDST_ACT_MISSING);
+        jvStatus = rpcError (rpcDST_ACT_MALFORMED);
         return PFR_PJ_INVALID;
     }
 
-    if (jvParams.isMember (jss::destination_amount))
+    if (! amountFromJsonNoThrow (
+        saDstAmount, jvParams[jss::destination_amount]))
     {
-        if (! amountFromJsonNoThrow (
-                saDstAmount, jvParams[jss::destination_amount]) ||
-            (saDstAmount.getCurrency ().isZero () &&
-             saDstAmount.getIssuer ().isNonZero ()) ||
-            (saDstAmount.getCurrency () == badCurrency ()) ||
-            saDstAmount <= zero)
-        {
-            jvStatus = rpcError (rpcDST_AMT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError (rpcDST_AMT_MALFORMED);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    convert_all_ =
+        saDstAmount == STAmount(saDstAmount.issue(), 1u, 0, true);
```

Comment:
`convert_all_` has no default value. Presumably every code path goes through this function, which will initialize it, but the uninitialized member makes me nervous.